src/instructlab/sdg/llmblock.py

```diff
@@ -81,8 +84,18 @@ def __init__(
         self.parser_name = parser_kwargs.get("parser_name", None)
         self.parsing_pattern = parser_kwargs.get("parsing_pattern", None)
         self.parser_cleanup_tags = parser_kwargs.get("parser_cleanup_tags", None)
+        # max_num_tokens should only be applicable to knowledge blocks
+        # gen_knowledge if the full/simple pipeline's knowledge generation block
+        if block_name != "gen_knowledge":
+            logger.debug(
+                f"Not applying max_num_tokens to block {block_name}. This is only applicable for gen_knowledge."
+            )
+            max_num_tokens = DEFAULT_MAX_NUM_TOKENS
         self.gen_kwargs = self._gen_kwargs(
-            gen_kwargs, model=self.ctx.model_id, temperature=0, max_tokens=4096
+            gen_kwargs,
+            model=self.ctx.model_id,
+            temperature=0,
+            max_tokens=max_num_tokens,
```

Comment:
I'm not sure this is doing what's intended, as the `max_tokens` here is only used as a default if none is specified in the config. But, we specify 2048 in the full and simple configs, which means we won't ever use the `max_tokens` value shown here.

Reply:
made this into `max_num_tokens_overwrite` as a position arg, if this arg != the default, we override the value already in gen_kwargs